spec.emu

```diff
@@ -4,20 +4,2187 @@
+            </emu-figure>
+
+            <p>As in the previous example, calling _A_.Evaluate() calls InnerModuleEvaluation on _A_, InnerAsyncSubgraphsEvaluation on _B_, and InnerModuleEvaluation on _C_, transitioning their [[Status]] respectively to ~evaluating-async~, ~async-subgraphs-evaluating-async~, and ~evaluating-async~. When InnerModuleEvaluation iterates over _C_'s dependencies, it finds _A_ and sets _A_.[[CycleRoot]] = _B_.[[CycleRoot]] = _C_.[[CycleRoot]] = _A_. At this point, the fields of the modules are given in <emu-xref href="#table-module-graph-deferred-async-cycle-root-fields-1"></emu-xref>:</p>
+
+            <emu-table id="table-module-graph-deferred-async-cycle-root-fields-1" class="module-overflow" caption="Module fields after the initial Evaluate() call">
+              <table>
+                <thead>
+                  <tr>
+                    <th>Module</th>
+                    <th>[[Status]]</th>
+                    <th>[[AsyncEvaluation]]</th>
+                    <th>[[AsyncParentModules]]</th>
+                    <th>[[PendingAsyncDependencies]]</th>
+                    <th>[[CycleRoot]]</th>
+                  </tr>
+                </thead>
+                <tr>
+                  <th>_A_</th>
+                  <td>~evaluating-async~</td>
+                  <td>*true*</td>
+                  <td>« »</td>
+                  <td>1 (_B_)</td>
+                  <td>_A_</td>
+                </tr>
+                <tr>
+                  <th>_B_</th>
+                  <td>~async-subgraphs-evaluating-async~</td>
+                  <td>*true*</td>
+                  <td>« _A_ »</td>
+                  <td>1 (_C_)</td>
+                  <td>_A_</td>
+                </tr>
+                <tr>
+                  <th>_C_</th>
+                  <td>~evaluating-async~</td>
+                  <td>*true*</td>
+                  <td>« _B_ »</td>
+                  <td>0</td>
+                  <td>_A_</td>
+                </tr>
+              </table>
+            </emu-table>
+
+            <p>Assume now that _C_ succesfully finishes evaluating. When that happens AsyncModuleExecutionFulfilled(_C_) is called, it sets _C_.[[Status]] to ~evaluated~ and _B_.[[Status]] to ~async-singraphs-evaluated~, and it performs _A_.ExecuteModule(). As stated, _A_'s execution causes a call to _B_.Evaluate(), which calls InnerModuleEvaluation on _B_ and on _C_. Since _C_.[[Status]] is ~evaluated~, _B_'s evaluation must not be blocked by the evaluation of the original cycle that _C_ is part of (_A_), thus InnerModuleEvaluation resets _C_.[[CycleRoot]] to _C_ and returns. InnerModuleEvaluation then synchronously calls _B_.ExecuteModule() and, assuming it completes succesfully, transitions _B_.[[Status]] to ~evaluated~. At this point, the fields of the modules are given in <emu-xref href="#table-module-graph-deferred-async-cycle-root-fields-2"></emu-xref>:</p>
```

Comment:
```suggestion
            <p>Assume now that _C_ succesfully finishes evaluating. When that happens AsyncModuleExecutionFulfilled(_C_) is called, it sets _C_.[[Status]] to ~evaluated~ and _B_.[[Status]] to ~async-subgraphs-evaluated~, and it performs _A_.ExecuteModule(). As stated, _A_'s execution causes a call to _B_.Evaluate(), which calls InnerModuleEvaluation on _B_ and on _C_. Since _C_.[[Status]] is ~evaluated~, _B_'s evaluation must not be blocked by the evaluation of the original cycle that _C_ is part of (_A_), thus InnerModuleEvaluation resets _C_.[[CycleRoot]] to _C_ and returns. InnerModuleEvaluation then synchronously calls _B_.ExecuteModule() and, assuming it completes succesfully, transitions _B_.[[Status]] to ~evaluated~. At this point, the fields of the modules are given in <emu-xref href="#table-module-graph-deferred-async-cycle-root-fields-2"></emu-xref>:</p>
```